python/cudf/cudf/_libxx/rolling.pyx

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+
+from __future__ import print_function
+import cudf
+import pandas as pd
+import numba
+
+from cudf._libxx.lib cimport *
+from cudf._libxx.column cimport *
+
+from cudf._libxx.aggregation cimport *
+
+from cudf._libxx.includes.rolling cimport (
+    rolling_window as cpp_rolling_window
+)
+
+
+def rolling(Column source_column, window, min_periods, center, op):
+    """
+    Rolling on input executing operation within the given window for each row
+
+    Parameters
+    ----------
+    source_column : input column on which rolling operation is executed
+    window : Size of the moving window, can be integer or numba DeviceArray
+    min_periods : Minimum number of observations in window required to have
+                  a value (otherwise result is null)
+    center : Set the labels at the center of the window
+    op : operation to be executed, as of now it supports MIN, MAX, COUNT, SUM,
+         MEAN and UDF
+
+    Returns
+    -------
+    A Column with rolling calculations
+    """
+
+    cdef size_type c_min_periods = min_periods
+    cdef size_type c_window = 0
+    cdef size_type c_forward_window = 0
+    cdef unique_ptr[column] c_result
+
+    if isinstance(window, numba.cuda.cudadrv.devicearray.DeviceNDArray):
```

Comment:
Can we just pass the window as a column instead of a numba devicearray?

Reply:
In that case we have to maintain two arguments, one for integer and one for array like, which looks odd.